File: ProjectEuler/Solver3.cs
namespace ProjectEuler
{
    class Solver3 : ISolver
    {
        public int Solve()
        {
            return Calculate(600851475143, 2);
        }

        static int Calculate(long i, int divisor)
        {
            return (i == divisor) ? divisor : i%divisor == 0 ? Calculate(i/divisor, divisor) : Calculate(i, divisor + 1);
        }
    }
}

File: C/Solver3.c
int calculate3(__int64 i, int divisor)
{
	while(i > divisor){
		if (i%divisor == 0) i /= divisor;
		else divisor++;
	}

	return divisor;
}

int Solver3()
{
    return calculate3(600851475143, 2);
}

        